.github/workflows/integration-tests.yaml

```diff
@@ -33,44 +33,82 @@ jobs:
         working-directory: ./
 
-      - name: Start DMSS
-        id: dmss
+      - name: Pull docker images
+        run: docker-compose pull
+        working-directory: example/
+
+      - name: Capture output DMSS version
         run: |
-          docker-compose pull
           docker-compose run --rm dmss cat src/version.txt > dmss_version.txt
           dmss_version=$(tail -n 1 dmss_version.txt)
+          echo "dmss_version=$dmss_version" >> "$GITHUB_OUTPUT"
+        working-directory: example/
+
+      - name: Capture job-api version
+        run: |
           docker-compose run --rm job-api cat version.txt > job_version.txt
           job_version=$(tail -n 1 job_version.txt)
-          docker-compose up -d && sleep 60 && docker-compose run --rm dmss reset-app
-          echo "dmss_version=$dmss_version" >> "$GITHUB_OUTPUT"
           echo "job_version=$job_version" >> "$GITHUB_OUTPUT"
         working-directory: example/
 
-      - name: Download test blueprints
-        run: |
-          npm pack @development-framework/dm-core-plugins
-          tar -xzf development-framework-dm-core-plugins-*.tgz
-          # reset-all.sh script is hardcoded to find it under node_modules
-          mkdir -p node_modules/@development-framework/
-          mv package/ node_modules/@development-framework/dm-core-plugins
+      - name: Run docker images
+        run: docker-compose up -d
         working-directory: example/
 
-      - name: Build and install dm-cli package
-        id: dm_cli
+      - name: Run reset app script
         run: |
-          python -m venv .venv
-          source .venv/bin/activate
-          pip install dm-cli
-          echo "version=$(dm --version)" >> "$GITHUB_OUTPUT"
+          sleep 60
+          docker-compose run --rm dmss reset-app
+        working-directory: example/
+
+      - name: Pack core plugins
+        run: npm pack @development-framework/dm-core-plugins
+        working-directory: example/
+
+      - name: Extract tarball
+        run: tar -xzf development-framework-dm-core-plugins-*.tgz
+        working-directory: example/
+
+      - name: Create required directory in node_modules
+        run: mkdir -p node_modules/@development-framework/
+        working-directory: example/
+
+      - name: Move package to directory in node_modules
+        run: mv package/ node_modules/@development-framework/dm-core-plugins
+        working-directory: example/
+
+      - name: Setup Python virtual environment
+        id: setup_venv
+        run: python -m venv .venv
+        working-directory: example/
+
+      - name: Activate Python virtual environment
+        run: source .venv/bin/activate
+        working-directory: example/
+
+      - name: Install dm-cli package
+        run: pip install dm-cli
+        working-directory: example/
+
+      - name: Capture dm-cli version
+        id: capture_version
+        run: echo "version=$(dm --version)" >> "$GITHUB_OUTPUT"
```

Comment:
I don't think this will work. https://stackoverflow.com/a/74669486/8231275 explains why. That's why I reactived the venv in the reset-all step below.
However, now I think the third approach makes most sense. Although I'm not sure why we would need a venv here at all. The code is already containerized.

Reply:
I agree that we don't need a venv in github actions. 
I removed the setup of the venv.